onyo/lib/onyo.py

```diff
@@ -266,6 +263,41 @@ def _generate_commit_message_subject(
         # return the shortest possible version of the commit message as fallback
         return msg
 
+    def sanitize_path(self, path: Union[Path, str]) -> Path:
+        """
+        Expects a relative or absolute path inside inside the repository.
+        These paths must not exist, just lead into the repo, so that the
+        function can be used to get paths to create new assets or move to
+        another location.
+
+        Returns an absolute `Path`.
+        """
+        onyo_path = Path(path).resolve()
+        if not onyo_path.is_absolute():
+            onyo_path = Path(Path.cwd(), path).resolve()
+
+        # TODO: raise a more reasonable error
+        if not onyo_path.is_relative_to(onyo.root):
+            raise ValueError("path is not relative to root.")
+
+        return onyo_path
+
+    def relative_to_root(
+            self, paths: Union[set[Path], list[Path]]) -> set[Path]:
+        """
+        Expects a list or set of absolute paths in the repository and returns a
+        set of paths relative to the repository root.
+        """
+        return {x.relative_to(self.root) for x in paths}
```

Comment:
I agree anyways, and changed it to just accept sets, because that is what we work with so far, and I think until we discuss the types we want to handle were, we should start with being more specific, as long as it is not clear that something more specific is needed!
So thanks for bringing up the set/list and str/Path issue :D